mixins/createGroupRouteMixin.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * A function to create a mixin to be used where there is an
+ * optional group param in the route.
+ *
+ * Designed to be used with the GroupSelect component.
+ *
+ * @param key used for the remember key and the route segment
+ * @returns a Vue mixin
+ */
+export default function createGroupRouteMixin(key) {
+  const rememberId = key
+  function routePath(id) {
+    return `/${key}/` + (id === 0 ? '' : id)
+  }
+  return {
+    computed: {
+      groupid: {
+        get() {
+          return this.$route.params.id ? parseInt(this.$route.params.id) : 0
+        },
+        set(val) {
+          const oldVal = this.groupid
+          val = val || 0
+          if (val !== oldVal) {
+            // We have changed the groupid away from the one in the route! Redirect...
+            this.$router.push(routePath(val))
+            if (val === 0) {
+              this.$store.commit('group/forget', {
+                id: rememberId
+              })
+            }
+          }
+        }
+      }
+    },
+    beforeDestroy() {
+      if (this._unwatchGroupRemember) this._unwatchGroupRemember()
+    },
+    created() {
+      this._unwatchGroupRemember = this.$store.watch(
+        (state, getters) => getters['group/remembered'](rememberId),
+        (val, oldVal) => {
+          if (oldVal === undefined && val === undefined && this.groupid !== 0) {
```

Comment:
This logic is a bit gnarly, and hard to understand I think....

Due to using `immediate: true` below, we get called _immediately_ (clever naming huh!) with the values, so we don't need to make a seperate call to get the remember value first....